src/server/scripts/Spells/spell_rogue.cpp

```diff
@@ -921,6 +930,48 @@ class spell_rog_serrated_blades : public SpellScriptLoader
     }
 };
 
+// 76297 - Weapon Skills
+class spell_rog_weapon_skills : public SpellScriptLoader
+{
+public:
+    spell_rog_weapon_skills() : SpellScriptLoader("spell_rog_weapon_skills") { }
+
+    class spell_rog_weapon_skills_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_rog_weapon_skills_SpellScript);
+
+        void HandleCast()
+        {
+            Player *player = GetCaster()->ToPlayer();
```

Comment:
You should add a Load() override that checks if caster is player because it would crash if the spell was cast by non-players:

```c++
            bool Load() override
            {
                return GetCaster()->GetTypeId() == TYPEID_PLAYER;
            }
```